### PR TITLE
templates: add ui for possible matches

### PR DESCRIPTION
**Pull request by Dinika**
Signed-off-by: Dinika <dinika.saxena@cern.ch>

### CODE REVIEW

File: src/inspirehep-holdingpen-js/templates/inspirehep-holdingpen-js/results/results.html [hunk 1]
```diff
@@ -97,6 +97,50 @@
               </p>
             </div>
 
+            <div ng-if="record._source._extra_data.record_matches.length > 1 
+            && record._source._workflow.data_type === 'hep' " ng-init="showPossibleMatches = true">
+                <div ng-show="showPossibleMatches">
+                    <div ng-if="record._source._extra_data.record_matches.length <= 5">
```
Comment (ammirate) [inline src/inspirehep-holdingpen-js/templates/inspirehep-holdingpen-js/results/results.html:103]: The matched identifiers will be from now in ``record._source._extra_data.matches.fuzzy``

File: src/inspirehep-holdingpen-js/templates/inspirehep-holdingpen-js/results/results.html [hunk 2]
```diff
@@ -97,6 +97,50 @@
...
+                          <a class="custom-control-description" ng-click="redirect('/literature/' + match)">
+                              labs.inspirehep.net/literature/{{match}}
+                          </a>
+                        </label>
+                    </div>
+                    <button class="btn btn-primary"
+                      ng-click="setDecision(match,'accept')"> Best Match
+                    </button>
+                    <a ng-click="showPossibleMatches = false">
+                      No Match Found
```
Comment (ammirate) [inline src/inspirehep-holdingpen-js/templates/inspirehep-holdingpen-js/results/results.html:123]: "None of these" would sound better IMO :)

**Review by ammirate — CHANGES_REQUESTED** [review]
Apart from the new field for the fuzzy matches ids, we have to add the logic to the button to resume the workflow.

**Comment by ammirate** [comment]
For the button that resumes the workflow, what it should do is:
* prepare the body of the request as ``{'match_recid': id}``
* send a POST request to ``'/api/holdingpen/<<WORKFLOW_ID>>/action/resolve'`` with the body defined above.

File: src/inspirehep-holdingpen-js/templates/inspirehep-holdingpen-js/results/results.html [hunk 3]
```diff
@@ -97,6 +97,50 @@
...
+                <div ng-show="showPossibleMatches">
+                    <div ng-if="record._source._extra_data.record_matches.length <= 5">
+                        <p class="small-text">Showing possible matches</p>
+                    </div>
+                    <div ng-if="record._source._extra_data.record_matches.length > 5">
+                        <p class="small-text">Showing 5 out of {{ (record._source._extra_data.record_matches.length-1) }} possible matches</p>
+                    </div>
+                    <div class="matchingRecords" ng-repeat="match in record._source._extra_data.record_matches">
+                        <label>
+                          <input id="{{match}}" name="possible-matches" type="radio">
```
Comment (drjova) [inline src/inspirehep-holdingpen-js/templates/inspirehep-holdingpen-js/results/results.html:111]: hmm shouldn't be ``<input ng-model="match" ... />``? (Maybe I'm missing something ☺️)

File: src/inspirehep-holdingpen-js/templates/inspirehep-holdingpen-js/results/results.html [hunk 4]
```diff
@@ -97,6 +97,50 @@
...
             </div>
 
+            <div ng-if="record._source._extra_data.record_matches.length > 1 
+            && record._source._workflow.data_type === 'hep' " ng-init="showPossibleMatches = true">
+                <div ng-show="showPossibleMatches">
+                    <div ng-if="record._source._extra_data.record_matches.length <= 5">
+                        <p class="small-text">Showing possible matches</p>
+                    </div>
+                    <div ng-if="record._source._extra_data.record_matches.length > 5">
+                        <p class="small-text">Showing 5 out of {{ (record._source._extra_data.record_matches.length-1) }} possible matches</p>
```
Comment (drjova) [inline src/inspirehep-holdingpen-js/templates/inspirehep-holdingpen-js/results/results.html:107]: IMHO we should remove the ``-1`` in ``record._source._extra_data.record_matches.length``. Shall we also remove  the hardcoded ``5``? ☺️

**Pull request closed without merging**